### ISSUE TITLE
SQL: IS NULL and IS NOT NULL doesn't behave correctly when used in HAVING clause

### ISSUE THREAD

**Issue by matriv** [elastic/elasticsearch#35139]
The following queries don't return the correct result:
```
SELECT max(languages) AS maxl, gender FROM "test_emp" WHERE emp_no IN(10018, 10019, 10020, 10021) GROUP BY gender HAVING maxl IS NULL;
```

```
SELECT max(languages) AS maxl, gender FROM "test_emp" WHERE emp_no IN(10018, 10019, 10020, 10021) GROUP BY gender HAVING maxl IS NOT NULL;
```

The issue is spotted in the IsNotNullProcessor since the value returned by the aggregation is a `Double.NaN` instead of a null.

**Comment by matriv** [comment]
@elastic/es-ql

**Comment by bpintea** [comment]
The issue with `NULL` filtering in `HAVING` seems to impact the other operators too. For instance, `int` fields with `NULL`s will behave incorrectly with comparisons; ex: `...HAVING SUM(int_field) < 2` will also return the fields with the sum `NULL`.

**Comment by bpintea** [comment]
> The issue with NULL filtering in HAVING seems to impact the other operators too. For instance, int fields with NULLs will behave incorrectly with comparisons; ex: ...HAVING SUM(int_field) < 2 will also return the fields with the sum NULL.

Opened separate #71645, as a different cause.

**Comment by bpintea** [comment]
Issue has been addressed with #71525.
